applications/matter_weather_station/README.rst

```diff
@@ -90,6 +103,7 @@ This application supports the following build types:
 
 * ``debug`` - Debug version of the application. You can use this version to enable additional features for verifying the application behavior, such as logs or command-line shell.
 * ``release`` - Release version of the application. You can use this version to enable only the necessary application functionalities to optimize its performance.
+* ``factory_data`` - Release version of the application that has factory data storage enabled. You can use this version to enable reading factory data necessary for Matter certification (such as product information, keys, and certificates) from a separate partition in the device non-volatile memory. To learn more about factory data read :doc:`matter:nrfconnect_factory_data_configuration` page in the Matter documentation.
```

Comment:
Add link to the enabling factory data section here?

Reply:
Ok, I rephrased this part and pre-pared factory data section, as I think my idea is not clear enough :) Using this type is just enough to enable factory data, but I added links to sections how to use ready factory data or generate new ones.